### PR TITLE
fix: [SRM-11719]: added tags to MS list view

### PR DESCRIPTION
**Pull request by karansaraswat19**
### Summary

Added tags to MS list view and increased MS name area by 1% while decreasing SLO/Error Budget area.

#### Screenshots

before:
Screenshot 2022-09-08 at 5 13 30 PM

after:
Screenshot 2022-09-08 at 5 20 12 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 4 Issues 
* Code issue 4 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11115)

**Comment by karansaraswat19** [comment]
trigger cypress

**Comment by karansaraswat19** [comment]
retrigger jest

**Comment by karansaraswat19** [comment]
retrigger titlecheck

**Comment by karansaraswat19** [comment]
retrigger coverage

**Comment by karansaraswat19** [comment]
trigger cypress

**Comment by karansaraswat19** [comment]
retrigger coverage

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx [hunk 1]
```diff
@@ -49,24 +50,34 @@ const RenderServiceName: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ r
...
-          className={css.monitoredServiceName}
-          title={monitoredService.serviceName}
-          font={{ align: 'left', size: 'normal', weight: 'semi-bold' }}
+      <Layout.Horizontal spacing="small" flex={{ alignItems: 'flex-start', justifyContent: 'start' }}>
+        <Link
+          to={routes.toCVAddMonitoringServicesEdit({
+            accountId,
+            orgIdentifier,
+            projectIdentifier,
+            identifier: monitoredService.identifier,
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:59]: destructure identifier from `monitoredService` at the top handling null check as well.

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx [hunk 2]
```diff
@@ -49,24 +50,34 @@ const RenderServiceName: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ r
...
-          {monitoredService.serviceName}
-        </Text>
-      </Link>
+          <Text
+            color={Color.PRIMARY_7}
+            className={css.monitoredServiceName}
+            title={monitoredService.serviceName}
+            font={{ align: 'left', size: 'normal', weight: 'semi-bold' }}
+          >
+            {monitoredService.serviceName}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:69]: destructure serviceName from `monitoredService` at the top handling null check as well.

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx [hunk 3]
```diff
@@ -49,24 +50,34 @@ const RenderServiceName: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ r
...
+          </Text>
+        </Link>
+        {monitoredService.tags && Object.keys(monitoredService.tags || {}).length ? (
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:72]: same.

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx [hunk 4]
```diff
@@ -49,24 +50,34 @@ const RenderServiceName: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ r
...
+          <TagsPopover
+            tags={monitoredService.tags}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:74]: same.

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx [hunk 5]
```diff
@@ -102,6 +102,15 @@ describe('Monitored Service list', () => {
     expect(getByText(getRiskLabelStringId(RiskValues.NEED_ATTENTION))).toBeDefined()
     expect(getByText(getRiskLabelStringId(RiskValues.HEALTHY))).toBeDefined()
     expect(document.title).toBe('cv.srmTitle | cv.monitoredServices.title | harness')
+    expect(
+      container.querySelectorAll('.TableV2--body [role="row"]')[0].querySelector('.bp3-popover-target')
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx:106]: Can we avoid targetting specific row ? Can we use some classname or test-id here ?

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx [hunk 6]
```diff
@@ -102,6 +102,15 @@ describe('Monitored Service list', () => {
...
+    ).toBeInTheDocument()
+    expect(
+      container
+        .querySelectorAll('.TableV2--body [role="row"]')[0]
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx:110]: same.

**Comment by karansaraswat19** [comment]
trigger cypress

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx [hunk 7]
```diff
@@ -102,6 +102,9 @@ describe('Monitored Service list', () => {
     expect(getByText(getRiskLabelStringId(RiskValues.NEED_ATTENTION))).toBeDefined()
     expect(getByText(getRiskLabelStringId(RiskValues.HEALTHY))).toBeDefined()
     expect(document.title).toBe('cv.srmTitle | cv.monitoredServices.title | harness')
+    // only differentiating factor for tags is its dark theme. So, used that here.
+    expect(container.querySelector('.bp3-dark')).toBeInTheDocument()
```
Comment (praneshg239) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx:106]: Please check in other places how they tested this scenario. I am not sure `.bp3-dark` would be a right option.

Comment (praneshg239) [reply src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx:106]: @harness/ui-cv any thought on this folks?

**Review by praneshg239 — COMMENTED** [review]
Please check using `.bp3-dark` in tests

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx (hunk 7 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx]: @praneshg239  suggested a fix for this , @karansaraswat19  Please update accordingly

**Comment by karansaraswat19** [comment]
trigger cypress

**Comment by karansaraswat19** [comment]
retrigger titlecheck

**Review by praneshg239 — COMMENTED** [review]
please check

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx [hunk 8]
```diff
@@ -102,6 +102,9 @@ describe('Monitored Service list', () => {
     expect(getByText(getRiskLabelStringId(RiskValues.NEED_ATTENTION))).toBeDefined()
     expect(getByText(getRiskLabelStringId(RiskValues.HEALTHY))).toBeDefined()
     expect(document.title).toBe('cv.srmTitle | cv.monitoredServices.title | harness')
+
```
Comment (karansaraswat19) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test.tsx:105]: @praneshg239 here is the test for tag

**Review by praneshg239 — DISMISSED** [review]

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx [hunk 9]
```diff
@@ -43,45 +43,55 @@ const CategoryProps: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ row }
 
 const RenderServiceName: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ row }) => {
   const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
-  const monitoredService = row.original
+  const { identifier, serviceName, tags, environmentRefList, type, environmentRef } = row.original || {}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:46]: do we need to add the default values for the destructured fields ?

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx [hunk 10]
```diff
@@ -43,45 +43,55 @@ const CategoryProps: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ row }
...
-          className={css.monitoredServiceName}
-          title={monitoredService.serviceName}
-          font={{ align: 'left', size: 'normal', weight: 'semi-bold' }}
+      <Layout.Horizontal spacing="small" flex={{ alignItems: 'flex-start', justifyContent: 'start' }}>
+        <Link
+          to={routes.toCVAddMonitoringServicesEdit({
+            accountId,
+            orgIdentifier,
+            projectIdentifier,
+            identifier: identifier,
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:58]: NIT: 
`accountId,
  orgIdentifier,
  projectIdentifier,
  identifier`

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx [hunk 11]
```diff
@@ -43,45 +43,55 @@ const CategoryProps: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ row }
...
+          />
+        ) : null}
+      </Layout.Horizontal>
       <Link
         to={routes.toCVAddMonitoringServicesEdit({
           accountId,
           projectIdentifier,
           orgIdentifier,
-          identifier: monitoredService.identifier,
+          identifier: identifier,
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:85]: Simply use 
`  accountId,
    projectIdentifier,
    orgIdentifier,
    identifier,`

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx [hunk 12]
```diff
@@ -43,45 +43,55 @@ const CategoryProps: Renderer<CellProps<MonitoredServiceListItemDTO>> = ({ row }
...
+          <Text
+            color={Color.PRIMARY_7}
+            className={css.monitoredServiceName}
+            title={serviceName}
+            font={{ align: 'left', size: 'normal', weight: 'semi-bold' }}
+          >
+            {serviceName}
+          </Text>
+        </Link>
+        {tags && Object.keys(tags || {}).length ? (
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:71]: Simply `!isEmpty(tags)` should take care of all checks.

File: src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx (hunk 9 above)
Comment (karansaraswat19) [reply src/modules/85-cv/pages/monitored-service/CVMonitoredService/components/MonitoredServiceListView/MonitoredServiceListView.tsx:46]: @harshilbits I just destructured it, I don't know if default values might be required somewhere

**Review by praneshg239 — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**